### PR TITLE
IMRPhenomXPHM implementation in jax (Accuracy benchmark)

### PR DESCRIPTION
**Pull request by narolaharsh**
## A working implementation of IMRPhenomXPHM in ripple
The implementation currently uses MSA method to comptue Euler angles. Other ways to compute the Euler angles (SpinTaylor and NNLO) are not yet implemented. 

### Mismatch comparison between lalsimulation and ripple 

image

Use [mismatch.py](https://github.com/GW-JAX-Team/ripple/blob/full-XPHM-dev-branch/tests/xphm/mismatch.py) to re-produce the plot above.

The mismatch comparison specifically use the MSA method of computing Euler angles. I tweaked the lalsimulation to force use MSA method (avoid switching between different methods). To re-produce the exact plot above, the [[mismatch.py](https://github.com/GW-JAX-Team/ripple/blob/full-XPHM-dev-branch/tests/xphm/mismatch.py) to generate mismatch plot] script needs to be runs in this spceific branch of lalsimulation. Otherwise, we might encounter slightly higher mismatches. 

Also adding two more testing scripts. 
1. A [gpu_timing.py](https://github.com/GW-JAX-Team/ripple/blob/full-XPHM-dev-branch/tests/xphm/gpu_timing.py) script to  test the speed. 
2. A [unit_test_xphm.py](https://github.com/GW-JAX-Team/ripple/blob/full-XPHM-dev-branch/tests/xphm/unit_test_xphm.py) script to generate a single XPHM waveform and make a comparison plot between lalsimulation and ripple.  

## Summary by CodeRabbit

* **New Features**
  * Full IMRPhenomXPHM waveform with precession and higher‑order‑mode support; MSA‑based precession initialisation and spin‑evolution utilities.
  * New elliptic‑integral and spin‑weighted spherical‑harmonic utilities.
* **Enhancements**
  * Extended IMRPhenomD phase with extra tunable parameters and added an amplitude variant without cutoff.
  * Exported additional physical constants and a typing alias.
* **Tests / Chores**
  * New XPHM comparison/mismatch and unit tests, benchmark formatting tweaks, and tightened pre‑commit file filtering.

### CODE REVIEW

File: src/ripplegw/waveforms/LALSimIMRPhenomX_precession.py [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+    mass_2: lighter black hole mass in solar mass units
+    chi1x, chi1y, chi1z: dimensionless spins of first black hole
+    chi2x, chi2y, chi2z: dimensionless spins of second black hole
+    emm: the index m of the mode (l, m)
+    reference_frequency:
+    kappa:
+    phiJ_Sf:
+    """
+    inspiral_mask = Mf < 0.3
+    # Mf = jnp.clip(Mf, a_max=0.3)
```
[the remark this answers is not in the document]
Comment (narolaharsh) [reply src/ripplegw/waveforms/LALSimIMRPhenomX_precession.py]: This follows the lalsimulation implementation, i.e, lalsimulation generates waveforms only up to 0.3 units in dimension less frequency. The dimension less frequecy is Mtotal * MTSUN_SI * frequency_array.

File: src/ripplegw/waveforms/spherical_harmonics.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                jnp.where(
+                    m == 1,
+                    jnp.sqrt(5.0 / (16.0 * jnp.pi))
+                    * jnp.sin(theta)
+                    * (1.0 + jnp.cos(theta)),
+                    jnp.sqrt(5.0 / (64.0 * jnp.pi)) * (1.0 + jnp.cos(theta)) ** 2,
+                ),
+            ),
+        ),
+    )
```
[the remark this answers is not in the document]
Comment (narolaharsh) [reply src/ripplegw/waveforms/spherical_harmonics.py]: The parent function makes sure that m value varies between minus ell to ell.
